### ISSUE TITLE
`\Psl\Filesystem\get_extension()` can return empty string

### ISSUE THREAD

**Issue by dragosprotung** [azjezz/psl#349]
**Describe the bug**
`\Psl\Filesystem\get_extension()` can return empty string

**To Reproduce**
```php
var_dump(\Psl\Filesystem\get_extension('test.')); // string(0) ""
```

**Expected behavior**
`null` to be returned.

**Additional context**
Not sure how this should be fixed.
One  aproach to fix this would be to change the return from `non-empty-string|null` to `string|null`.
Personally I would expect `null` to be returned for a file without an extension but with a `.` in the name (like in the example above).

I can make a PR once the direction is decided.

**Comment by azjezz** [comment]
I'm leaning more toward returning null in this case.

**Comment by dragosprotung** [comment]
Closed by #350
